src/App.tsx

```diff
@@ -1,61 +1,110 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TodoWithUser } from './types/TodoWithUser';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const findUserById = (userId: number) => (
+  usersFromServer.find(user => user.id === userId) || null
+);
+const todosWithUsers: TodoWithUser[] = todosFromServer.map(todo => {
+  const currentUser = findUserById(todo.userId);
+
+  return {
+    ...todo,
+    user: currentUser,
+  };
+});
+
+export const App: React.FC = () => {
+  const [todosList, setTodosList] = useState<TodoWithUser[]>(todosWithUsers);
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [titleValue, setTitleValue] = useState('');
+  const [submitTouched, setSubmitTouched] = useState(false);
+
+  const resetInputs = () => {
+    setTitleValue('');
+    setSelectedUser(0);
+    setSubmitTouched(false);
+  };
+
+  const onFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!selectedUser || !titleValue) {
+      setSubmitTouched(true);
+    } else {
```

Comment:
Return from function immediately if you have errors. Don't use else in this case.
Also you need to check if your title is blank string only with spaces. Use trim()